streaming/index.js

```diff
@@ -231,6 +231,10 @@ const startServer = async () => {
     help: 'The number of Redis channels the streaming server is subscribed to',
   });
 
+  // When checking metrics in the browser, the favicon is requested this
+  // prevents the request from falling through to the API Router, which would
+  // error for this endpoint:
+  app.get('/favicon.ico', (req, res) => res.end());
```

Comment:
I am not sure I understand. Could we have this return a 404, like any un-handled path?

Reply:
We could, but browsers usually prefer a 200 response on favicons, even if they're no content responses?

could also just fs.readFile the favicon from public, but that seemed a bit excessIve

Reply:
I mean, I can certainly make it return a favicon or return a 404 — but the 404 would likely appear in error logs, and maybe alert on monitoring tools, so 200 or 201 is probably better.

Reply:
Which error logs? If it's the browser, it's not really relevant, as a normal user won't access the streaming server directly that way.

Reply:
nginx & load balancers.

Reply:
Right. But that doesn't sound like much of an issue, few people would open the streaming endpoints in a browser.

Reply:
hm, I don't understand how it would end up into nginx & load balancers if it's only ever accessed directly (not under `/streaming`); I still think this does not need any special handling whatsoever, but if we do some special handling, it should return 404 or a valid icon

Reply:
Keep in mind that falling through to the rest of the code invokes database connections, and a bunch of complex logic that has assumptions on being a request for `/api/v1/streaming/*`, given we can't terminate connections based on the Accept header not indicating that text/event-stream (eventsource) is desired: https://github.com/mastodon/mastodon/pull/26299/files#r1284439606

If we could do that, then we'd have a simpler code path through this for eventsource streaming connections, because we'd be able to go "oh, this is a request for something that's not an `text/event-stream` response, reject the request, but as we can't the request will make it all the way through the authentication & authorization code which has certain assumptions and it'll end up logged as an error during development, which makes developing the streaming server confusing.

It's a single line of code, which prevents an error from occurring during development or requests from the browser directly against the streaming server. I don't see the harm in having this, and not sure why this has generated so much discussion given the state of the rest of the streaming code.